### ISSUE TITLE
Is this thread safe?

### ISSUE THREAD

**Issue by dersoz** [GeoDienstenCentrum/sass-maven-plugin#209]
Are there any plans to make this plugin marked as threadSafe? Is it safe to use it in parallel maven build even though it is not marked as threadSafe? I am getting a warning from maven about this plugin's thread safety when I run maven in parallel with command `mvn package -T 2`

### Expected Behavior
No thread safety warnings from maven when used this plugin with the command `mvn package -T 2`

### Actual Behavior
```
[WARNING] * Your build is requesting parallel execution, but project      *
[WARNING] * contains the following plugin(s) that have goals not marked   *
[WARNING] * as @threadSafe to support parallel building.                  *
[WARNING] * While this /may/ work fine, please look for plugin updates    *
[WARNING] * and/or request plugins be made thread-safe.                   *
[WARNING] * If reporting an issue, report it against the plugin in        *
[WARNING] * question, not against maven-core                              *
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked @threadSafe in <myGroupId> - <myArtifactId>:
[WARNING] nl.geodienstencentrum.maven:sass-maven-plugin:3.5.5
[WARNING] Enable debug to see more precisely which goals are not marked @threadSafe.
[WARNING] *****************************************************************
```

### Environment and configuration
Maven version: 3.5.3
Java version: 1.8.0_172
Plugin version: 3.5.5

**Comment by mprins** [comment]
Your report is missing key information for debugging and or analysis. Please use the issue template provided.

**Comment by dersoz** [comment]
Hi Mark. I think I have provided the key information for the question above. This plugin is used in our codebase and we really need to know if this is threadsafe or not. Can you comment on my question?

**Comment by mprins** [comment]
I have no idea, I have no testcases that either fail of pass for this use case; the warning as such does mean it isn't it just means there is no annotation, adding the annotation will make the warning disappear, but doesn't make things magically threadsafe. 

So is you can provide a testcase, as in a maven project that fails or passes, there is something to work with, if you don't nothing will happen. 
If you need inspiration there are various projects in the codebase: https://github.com/GeoDienstenCentrum/sass-maven-plugin/tree/master/src/test/projects/

**Comment by lexbauer** [comment]
Hi Mark, it is difficult to test thread safety as thread behaviour is unpredictable and hard to reproduce. Instead of providing test cases, I propose to have me investigate it. Now, I already took a look at the **update stylesheets mojo** and I noticed most code (if not all) is already thread safe (barely using shared resources, shared resources are only getters). :-)
The only code that might be unsafe is AbstractSassMojo.extractBourbonResources(...). In here, we have a FileOutputStream. If you make extractBourbonResources(...) synchronized, then the **update stylesheets mojo** becomes safe for sure.
